toolkit/geoview-compose/src/main/java/com/arcgismaps/toolkit/geoviewcompose/GeoViewScope.kt

```diff
@@ -0,0 +1,499 @@
+            } else {
+                screenCoordinate.offset(offset)
+            }
+        }
+    }
+
+    /**
+     * Analogue of Layout which allows to sub-compose the Callout container during the measuring stage,
+     * and anchor the Callout leader at the given screenCoordinate. The Callout is drawn using a Box
+     * and by default it is anchored to its top left corner. In order to anchor the Callout at the tip
+     * of the leader we need to determine the size of its content to calculate the anchor point's
+     * location before drawing the Callout on the screen.
+     *
+     * @param leaderScreenCoordinate Represents the x,y coordinate for the location on GeoView
+     * @param maxSize The calculated maximum size of the callout container
+     * @since 200.5.0
+     */
+    @Composable
+    private fun CalloutSubComposeLayout(
+        modifier: Modifier = Modifier,
+        leaderScreenCoordinate: ScreenCoordinate,
+        maxSize: DpSize,
+        calloutContainer: @Composable () -> Unit
+    ) {
+        val configuration = LocalDensity.current
+        val maxWidthInPx = with(configuration) {
+            maxSize.width.roundToPx()
+        }
+        val maxHeightInPx = with(configuration) {
+            maxSize.height.roundToPx()
+        }
+
+        SubcomposeLayout(modifier = modifier) { constraints ->
+            // set the max width to the lesser of the available size or the maxWidth
+            val layoutWidth = Integer.min(constraints.maxWidth, maxWidthInPx)
+            // set the max height to the lesser of the available size or the maxHeight
+            val layoutHeight = Integer.min(constraints.maxHeight, maxHeightInPx)
+            // measure the content with the constraints
+            val calloutContainerPlaceable = subcompose(slotId = 0) {
+                calloutContainer()
+            }[0].measure(
+                constraints.copy(
+                    maxWidth = layoutWidth,
+                    maxHeight = layoutHeight
+                )
+            )
+            // The default (0,0) value is on the top-left edge of the callout container.
+            // This moves the anchor to the bottom-middle point using X,Y offsets,
+            // and ensures that the leader's anchor point always represents the tapped location,
+            // in this case the bottom-middle leader position.
+            val calloutOffsetX =
+                leaderScreenCoordinate.x.toInt() - (calloutContainerPlaceable.width / 2)
+            val calloutOffsetY =
+                leaderScreenCoordinate.y.toInt() - calloutContainerPlaceable.height
+            // place the callout in the layout
+            layout(layoutWidth, layoutHeight) {
+                calloutContainerPlaceable.place(calloutOffsetX, calloutOffsetY)
+            }
+        }
+    }
+
+    /**
+     * UI default properties for the [Callout] component.
+     */
+    private data class CalloutProperties(
```

Comment:
Seems like the way this is used, it should be an `object`?
```suggestion
    private object DefaultCalloutProperties(
```